model/system_metrics_test.go

```diff
@@ -571,6 +576,57 @@ func TestSystemMetricsDownload(t *testing.T) {
 		assert.Equal(t, 2, rawReader.batchSize)
 		assert.Equal(t, FileText, rawReader.format)
 	})
+	t.Run("WithPagination", func(t *testing.T) {
+		bucket, err := systemMetrics2.Artifact.Options.Type.Create(
+			ctx,
+			env,
+			conf.Bucket.SystemMetricsBucket,
+			systemMetrics2.Artifact.Prefix,
+			string(pail.S3PermissionsPrivate),
+			false,
+		)
+		require.NoError(t, err)
+		keys, data, err := GenerateSystemMetrics(ctx, bucket, 5)
+		require.NoError(t, err)
+		fullData := []byte{}
+		for _, key := range keys {
+			fullData = append(fullData, data[key]...)
+		}
+		systemMetrics2.Artifact.MetricChunks["pagination"] = MetricChunks{
+			Chunks: keys,
+			Format: FileText,
+		}
+		systemMetrics2.populated = true
+		systemMetrics2.Setup(env)
+
+		readData, idx, err := systemMetrics2.DownloadWithPagination(ctx, SystemMetricsDownloadOptions{
+			MetricType: "pagination",
+			PageSize:   33,
+		})
+		require.NoError(t, err)
+		assert.Equal(t, fullData[:64], readData)
```

Comment:
Can we add a comment saying that the 64 comes from expecting two 32-byte chunk strings to be read?

Reply:
Actually thinking on it, it might just have been clearer if `PageSize` was calculated as `len(chunk) + 1` and then this would check `fullData[:2 * len(chunk)]` (without needing a comment).